TL/18_Configuring_a_project._Property_files/Configuring_a_project.Property_files_README.md

```diff
@@ -0,0 +1,269 @@
+      "UserName": "",
+      "Password": "Testing123!"
+    },
+    "SupplierReadOnly": {
+      "UserName": "",
+      "Password": "Testing123!"
+    },
+    "BuyerAdmin": {
+      "UserName": "",
+      "Password": "Testing123!c"
+    },
+    "BuyerUser": {
+      "UserName": "",
+      "Password": "Testing123!c"
+    },
+    "BuyerReadOnly": {
+      "UserName": "",
+      "Password": "Testing123!c"
+    }
+  }
+}
+
+```
+
+## Property file readers
+
+In Java, there are several ways to read properties from property files. Some common types of property file readers
+include:
+
+1. *Properties Class*: Java's built-in Properties class can be used to read properties from a .properties file.
+   You can load properties from a file using methods like load(InputStream) or load(Reader).
+
+2. *ResourceBundle Class*: The ResourceBundle class can also be used to read properties from a .properties file.
+   It provides a way to access locale-specific resources in a flexible way.
+
+3. *FileReader Class*: You can use the FileReader class along with other input stream classes like BufferedReader
+   to read properties from a file.
+
+4. *Scanner Class*: The Scanner class can be used to parse input from various sources, including files.
+   You can use it to read properties from a file line by line.
+
+5. *Apache Commons Configuration*: Apache Commons Configuration library provides a powerful and flexible way
+   to read and write configuration data from various sources, including properties files.
+
+6. *Spring's PropertySource*: If you are using the Spring framework, you can use the @PropertySource annotation
+   along with Environment to read properties from property files.
+
+***
+
+### Properties Class load() method
+
+```java
+package utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class PropertiesManager {
+    private static final Logger logger = LogManager.getLogger(PropertiesManager.class);
+    private static String browser;
+    public static String homeUrl;
+    public static String registerUrl;
+    public static String overviewUrl;
+    public static String baseUrl;
+    private static Properties properties = new Properties();
+
+    private PropertiesManager() {
+        try (FileInputStream input = new FileInputStream("src/main/resources/config.properties")) {
+            properties.load(input);
+
+            browser = properties.getProperty("browser");
+            homeUrl = properties.getProperty("homeUrl");
+            registerUrl = properties.getProperty("registerUrl");
+            overviewUrl = properties.getProperty("overviewUrl");
+            baseUrl = properties.getProperty("baseUrl");
+
+        } catch (IOException e) {
+            logger.error("Error loading properties file: " + e.getMessage());
+        }
+    }
+
+    public static String getBrowser() {
```

Comment:
if here we read property from file i think a name like initProperty or something like that will be more appropriate by java naming convention. Name should reflect the action inside